### PR TITLE
Add Pre-IPM practice (#965)

### PR DESCRIPTION
**Pull request by mkollasch-vmware**

### CODE REVIEW

File: content/practices/pre-ipm/index.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+# remove "draft:true" when ready for final publication 
+draft: true
```
Comment (joemoore) [inline content/practices/pre-ipm/index.md:3]:
```suggestion
draft: false
```

Comment (joemoore) [reply content/practices/pre-ipm/index.md:3]: Please set to false to see the preview. I'm going to update the template to remove this, as it did not function as I expected.

File: content/practices/pre-ipm/index.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+# remove "draft:true" when ready for final publication 
+draft: false
+
+title: "Pre-IPM"
```
Comment (joemoore) [inline content/practices/pre-ipm/index.md:5]: I feel we need to spell out the title. Pre-Iteration Planning Meeting or similar.

**Comment by clhynfield** [comment]
To make the name of this practice more readily understood and discoverable, I'd like to suggest renaming it to replace the initialism, from "Pre-IPM" to "Iteration Pre-Planning". 

If IPM itself were a ubiquitous term in the software industry, like MVP, using it with a prefix might be quick to parse and easy to understand. "IPM" isn't an industry-recognized term, though – The definition "Iteration Planning Meeting" doesn't show up at all in the first twenty pages of Google search results for [IPM][], where [TDD][], [MVP][], and [I18N][] all return the software-industry definitions in the top three results. The only public reference I could find to "pre-IPM" was in a [StackExchange thread][] where most folks responded, "I've never heard of pre-IPM – what's that?" 

[IPM]: https://www.google.com/search?q=IPM
[TDD]: https://www.google.com/search?q=TDD
[MVP]: https://www.google.com/search?q=MVP
[I18N]: https://www.google.com/search?q=I18N
[StackExchange thread]: https://softwareengineering.stackexchange.com/questions/372401/what-is-the-difference-between-pre-ipm-and-product-backlog-refinement

**Comment by mkollasch-vmware** [comment]
I noticed odd behavior in the initial commit which was resolved seemingly accidentally in the most recent commit, which I've described in issue #1072.

File: content/practices/pre-ipm/index.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+linkTitle: "Pre-IPM"
+description: "An optional planning session to make the [Iteration Planning Meeting (IPM)](/practices/ipm) more productive if needed."
```
Comment (joemoore) [inline content/practices/pre-ipm/index.md]: Unfortunately we have a bug that does not allow these relative links to work properly. For now, please remove the link and we will add an Issue.

Comment (mkollasch-vmware) [reply content/practices/pre-ipm/index.md]: These are resolved in commit `041a95719f6b4e022e25dfbf7eef552c741be132`, and I took the liberty of opening issue #1072 to describe it (apologies if it's a duplicate).

**Comment by joemoore** [comment]
The spell-checker is failing. https://github.com/vmware-tanzu/tanzu-dev-portal/pull/1067/checks?check_run_id=3090003245

We need better documentation about this, but the spell checker is pretty primitive, especially regarding tech language. 

1. Please review the flagged words: https://github.com/vmware-tanzu/tanzu-dev-portal/pull/1067/checks?check_run_id=3090003245
1. If it's a real spelling mistake, please fix.
1. If the it has flagged a correctly spelled word or phrase, the checker might consider it uncommon or a colloquialism. Consider alternatives. 
1. If you would still like to use the word, it has flagged a name, tech-term, etc. , please edit `/cust.dict`. Note the regex-like syntax. You word might have already been added. Feel free to add your word(s) to the bottom of this file (a pre-commit hook will sort it).

File: content/practices/iteration-pre-planning/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+linkTitle: "Iteration Pre-Planning"
+description: "An optional planning session to make the Iteration Planning Meeting (IPM) more productive if needed."
+tags: ["Delivery"]
+length: "30 min - 1 hour"
+participants: "Product managers, 1-2 engineers"
+image: "default-cover.png"
+lastmod: "2021-07-15"
+date: "2021-07-14"
+why: 
+- If a team finds that an excessive amount of time in the [Iteration Planning Meeting (IPM)](/practices/ipm) is spent on revising stories before estimation, editing or grooming them in a smaller group beforehand can help the larger meeting produce better results.
```
Comment (joemoore) [inline content/practices/iteration-pre-planning/index.md:15]: Please remove this link as well due to #1072, which actually applies to all front-matter / metadata.

Comment (mkollasch-vmware) [reply content/practices/iteration-pre-planning/index.md]: This seems unrelated to #1072. Unlike the `description` section, the link in the `why` section renders correctly, but its route is resolved differently than the link in the main body of the article. Commit `9467191c85e9bcd13791fc3ccfa3f59e5042f2d6` should fix it.

File: content/practices/iteration-pre-planning/index.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+   Not every concern needs to be caught.
+   After all, one engineering pair can’t be expected to replicate the insights of an entire team.
+   Focus on quickly resolving the aspects most likely to be problematic or disruptive during IPM.
+
+   Discussion of how to avoid these problems in the future is appropriate as well.
+
+1. Repeat steps 2-4 for the remaining user stories that were prepared for this meeting.
+
+
+#### Success/Expected Outcomes
```
Comment (joemoore) [inline content/practices/iteration-pre-planning/index.md:59]:
```suggestion
## Success/Expected Outcomes
```

File: content/practices/iteration-pre-planning/index.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+You know you are done when all stories staged for IPM are ready to be estimated according to the engineer(s) present.
+You know it’s working when improvement is observed in the issues identified in Step 1.
+
+
+#### Facilitator Notes & Tips
```
Comment (joemoore) [inline content/practices/iteration-pre-planning/index.md:64]:
```suggestion
## Facilitator Notes & Tips
```

**Comment by joemoore** [comment]
@BrianMMcClain Ready to merge!

**Review by BrianMMcClain — APPROVED** [review]

**Pull request merged**